Documentation/Architecture.md

```diff
@@ -1,148 +1,140 @@
-into the pod and hits the specialize endpoint on the environment container. 
-This causes the function to be loaded. The pod is now specific to that 
-function. This function pod is cached; it's cleaned up if it's unused for a few minutes.
-
-PoolManager selects a generic pod from the warm pool, specializes it, 
-and recycles the pod if there are no further requests to the function after a few minutes.
-It makes PoolManager suitable for functions that are short-living
-and requires a short cold start time [1].
-
-However, PoolManager only selects one pod per function, which is not
-suitable for serving massive traffic. In such cases, you should consider
-using NewDeploy as executor type of function.
-
-[1] The cold start time depends on the package size of the function. If it's
-a snippet of code, the cold start time usually is less then 100ms.
-
-#### NewDeploy 
-
-NewDeploy creates deployment, service, and HPA for functions in order to handle 
-massive traffic.
-
-NewDeploy watches the function CRD changes and creates a Kubernetes deployment, 
-service, and HPA for a function. NewDeploy will scale the replicas of a function 
-deployment to the minimum feasible scale setting, if the minimum scale setting of 
-a function is greater than 0. The 'fetcher' inside the pod uses a URL in the 
-JSON payload, which is attached as a parameter to start fetcher, to download the 
-function package instead of waiting for calls from NewDeploy. 
-
-When a function experiences a traffic spike, the service helps to distribute the requests to 
-pods belonging to the function for better workload distribution and lower latency. Also, 
-the HPA scales the replicas of the deployment based on the conditions set by the user.
- 
-This approach though increases the cold time of a function, but also makes NewDeploy 
-suitable for functions designed to serve massive traffic.
+PoolManager watches the environment CRD changes and eagerly creates generic pools for environments.
+It uses Kubernetes deployments to do that.
+The environment container runs in a pod with the 'fetcher' container.
+Fetcher is a straightforward utility that downloads a URL sent to it and saves it at a configured location (shared volume).
+
+The implementation chooses a generic pod from the pool, relabels it to "orphan".
+The pod from the deployment invokes fetcher to copy the function into the pod and hits the specialize endpoint on the environment container.
+This causes the function to be loaded.
+The pod is now specific to that function.
+This function pod is cached; it's cleaned up if it's unused for a few minutes.
+
+PoolManager selects a generic pod from the warm pool, specializes it, and recycles the pod if there are no further requests to the function after a few minutes.
+It makes PoolManager suitable for functions that are short-living and requires a short cold start time [1].
+
+However, PoolManager only selects one pod per function, which is not suitable for serving massive traffic.
```

Comment:
This is not 100% true with RPP. Also Concurrency feature adds some more scenarios, but may be for some other PR